homeassistant/components/unifiprotect/models.py

```diff
@@ -37,92 +30,85 @@ class PermRequired(int, Enum):
-        )
-        object.__setattr__(self, "ufp_value", split_tuple(self.ufp_value))
-        object.__setattr__(self, "ufp_enabled", split_tuple(self.ufp_enabled))
-
     def get_ufp_value(self, obj: T) -> Any:
         """Return value from UniFi Protect device."""
-        if (ufp_value := self.ufp_value) is not None:
-            if TYPE_CHECKING:
-                # `ufp_value` is defined as a `str` in the dataclass, but
-                # `__post_init__` converts it to a `tuple[str, ...]` to avoid
-                # doing it at run time in `get_nested_attr` which is usually called
-                # millions of times per day. This tells mypy that it's a tuple.
-                assert isinstance(ufp_value, tuple)
-            return get_nested_attr(obj, ufp_value)
-        if (ufp_value_fn := self.ufp_value_fn) is not None:
-            return ufp_value_fn(obj)
-
-        # reminder for future that one is required
+        # ufp_value or ufp_value_fn is required, the
+        # RuntimeError is to catch any issues in the code
+        # with new descriptions.
         raise RuntimeError(  # pragma: no cover
             "`ufp_value` or `ufp_value_fn` is required"
         )
 
+    def has_required(self, obj: T) -> bool:
+        """Return if required field is set."""
+        return True
+
     def get_ufp_enabled(self, obj: T) -> bool:
-        """Return value from UniFi Protect device."""
-        if (ufp_enabled := self.ufp_enabled) is not None:
-            if TYPE_CHECKING:
-                # `ufp_enabled` is defined as a `str` in the dataclass, but
-                # `__post_init__` converts it to a `tuple[str, ...]` to avoid
-                # doing it at run time in `get_nested_attr` which is usually called
-                # millions of times per day. This tells mypy that it's a tuple.
-                assert isinstance(ufp_enabled, tuple)
-            return bool(get_nested_attr(obj, ufp_enabled))
+        """Return if entity is enabled."""
         return True
 
-    def has_required(self, obj: T) -> bool:
-        """Return if has required field."""
-        if (ufp_required_field := self.ufp_required_field) is None:
-            return True
-        if TYPE_CHECKING:
-            # `ufp_required_field` is defined as a `str` in the dataclass, but
-            # `__post_init__` converts it to a `tuple[str, ...]` to avoid
-            # doing it at run time in `get_nested_attr` which is usually called
-            # millions of times per day. This tells mypy that it's a tuple.
-            assert isinstance(ufp_required_field, tuple)
-        return bool(get_nested_attr(obj, ufp_required_field))
+    def __post_init__(self) -> None:
+        """Pre-convert strings to tuples for faster get_nested_attr.
+
+        get_ufp_value/has_required/get_ufp_enabled are overridden based
+        on what is defined in the dataclass so they only have to be
+        worked out once.
+        """
+
```

Comment:
The key change here is that instead of overriding the types from string to tuples, we build the functions instead so all the types line up nicely and we don't have to work out how the attribute is fetched every time `get_ufp_value`, `get_ufp_enabled`, or `has_required` is called since the function always knows what to do